### PR TITLE
Limit Comment to 300000 Chars

### PR DESCRIPTION
**Pull request by aweijun**
**DO NOT MERGE**

## Details

Limit Comment to 300000 Characters.

### CODE REVIEW

File: src/main/java/seedu/trackermon/model/show/Comment.java [hunk 1]
```diff
@@ -10,11 +17,19 @@
      * @param comment A valid comment.
      */
     public Comment(String comment) {
+        checkArgument(isValidComment(comment), MESSAGE_CONSTRAINTS);
         String emojiFilter = "[(\\u00a9|\\u00ae|[\\u2000-\\u3300]|\\ud83c[\\ud000-\\udfff]|"
-                + "\\ud83d[\\ud000-\\udfff]|\\ud83e[\\ud000-\\udfff])]";
+                        + "\\ud83d[\\ud000-\\udfff]|\\ud83e[\\ud000-\\udfff])]";
```
Comment (JonathanHoshi) [inline src/main/java/seedu/trackermon/model/show/Comment.java:22]: What is this awkward spacing for?

Comment (arcornior) [inline src/main/java/seedu/trackermon/model/show/Comment.java:22]: Also what does this regex match?

**Review by arcornior — COMMENTED** [review]
Should an updated UG be included in this PR too?

File: src/main/java/seedu/trackermon/model/show/Comment.java (hunk 1 above)
Comment (aweijun) [reply src/main/java/seedu/trackermon/model/show/Comment.java:22]: I will remove it in the next commit.

Comment (aweijun) [reply src/main/java/seedu/trackermon/model/show/Comment.java:22]: I am not sure what you mean?

**Review by JonathanHoshi — APPROVED** [review]
LGTM!

**Review by Ardentsoul — APPROVED** [review]
LGTM!

File: src/main/java/seedu/trackermon/model/show/Comment.java (hunk 1 above)
Comment (arcornior) [reply src/main/java/seedu/trackermon/model/show/Comment.java]: Wait never mind, I must've seen wrongly. I thought you added a new regex.

**Review by arcornior — CHANGES_REQUESTED** [review]
Since we're going to change how Comment works, I believe a UG update should be included in this PR! Great job catching this bug!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -616,7 +616,8 @@ So if you use `sort n/asc s/asc so/haha`, the required criteria are name and sta
 
 | Term                               | Description                                                                                                                                                                          |
 |------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| **AND search**                     | AND search finds all of the keywords.  For example, `find n/Shutter Island` returns only results that contain Shutter and Island.                                                    |
+| **AND search**                     | AND search finds all of the keywords.  For example, `find n/Shutter Island` returns only results that contain Shutter and Island.                |
+| **ASCII**                   | A 8-bit character set containing 128 characters. This includes number 0-9, upper and lower case english Letters (A-Z) and certain special characters.                |
```
Comment (arcornior) [inline docs/UserGuide.md:620]: Should this be `An 8-bit character set` instead?
Also, `This includes the numbers 0-9` instead?
Also, `English letters (A-Z), and` instead?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -170,7 +170,7 @@ In the example above , `find` is the **command word** while `n/` is the **prefix
...
 | RATING    | r/     | Rating must be a **whole number** from **0 to 5**.                                                                                                                                                                                                                                                                    |
-| COMMENT   | c/     | A comment can contain **any text**, but **emojis will be removed** from input.                                                                                                                                                                                                                                                                                                                      |
+| COMMENT   | c/     | A comment can only contain **ASCII values** and is limited to 300000 **ASCII values**.                                                                                                                                                                                                                                                                                                                      |
```
Comment (arcornior) [inline docs/UserGuide.md:173]: Maybe we can add a link to the glossary table for `ASCII`?
Also maybe we could rephrase it to `A comment can only contain up to 300000 ASCII values` instead?

File: docs/UserGuide.md (hunk 2 above)
Comment (arcornior) [reply docs/UserGuide.md:620]: Also should we shift ASCII's glossary entry to the bottom of the table? I don't think we should keep it in the middle of the search entries. Kind of breaks the flow if readers want to compare the 2 searches. Just my 2 cents!

**Review by arcornior — CHANGES_REQUESTED** [review]
Just a few changes!

**Review by arcornior — APPROVED** [review]
LGTM!
